crates/rome_js_analyze/src/analyzers/nursery/no_misused_new.rs

```diff
@@ -0,0 +1,232 @@
+    /// This rule triggers warnings in the following scenarios:
+    /// - When a class has a method named `new`.
+    /// - When an interface defines a method named `constructor` or `new` that returns the interface type.
+    /// - When a type alias has a `constructor` method.
+    ///
+    /// You should not use this rule if you intentionally want a class with a `new` method, and you're confident nobody working in your code will mistake it with an `constructor`.
+    ///
+    /// Source: https://typescript-eslint.io/rules/no-misused-new/
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```ts,expect_diagnostic
+    /// declare class C {
+    ///   new(): C;
+    /// }
+    /// ```
+    ///
+    /// ```ts,expect_diagnostic
+    /// interface I {
+    ///   new (): I;
+    ///   constructor(): void;
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```ts
+    /// declare class C {
+    ///   constructor();
+    /// }
+    ///
+    /// interface I {
+    ///   new (): C;
+    /// }
+    /// ```
+    ///
+    pub(crate) NoMisusedNew {
+        version: "next",
+        name: "noMisusedNew",
+        recommended: false,
+    }
+}
+
+declare_node_union! {
+    pub(crate) DeclarationQuery = TsInterfaceDeclaration |  TsTypeAliasDeclaration | JsClassDeclaration | TsDeclareStatement
+}
+
+pub(crate) enum RuleState {
+    ClassMisusedNew(TextRange),
+    InterfaceMisusedNew(TextRange),
+    InterfaceMisusedConstructor(TextRange),
+    TypeAliasMisusedConstructor(TextRange),
+}
+
+impl RuleState {
+    fn message(&self) -> MarkupBuf {
+        match self {
+            RuleState::ClassMisusedNew(_) => (markup! {
+                "Do not use the `new` method in class."
+            })
+            .to_owned(),
+            RuleState::InterfaceMisusedNew(_) => (markup! {
+                "Do not use the `new` method in interface."
+            })
+            .to_owned(),
+            RuleState::InterfaceMisusedConstructor(_) => (markup! {
+                "Do not use the `constructor` method in interface."
+            })
+            .to_owned(),
+            RuleState::TypeAliasMisusedConstructor(_) => (markup! {
+                "Do not use the `constructor` method in type alias."
+            })
+            .to_owned(),
+        }
+    }
+
+    fn note(&self) -> MarkupBuf {
+        match self {
+            RuleState::ClassMisusedNew(_) => (markup! {
+                "`new` is not typical in class."
+            })
+            .to_owned(),
+            RuleState::InterfaceMisusedNew(_) => (markup! {
+                "`new` is misleading in interface."
```

Comment:
We could use a more detailled message such as `new makes an interface a constructor. The returned type should certainly a different type from its constructor.`. Any thought?

Reply:
I agree with that. I'll update it more descriptive.